src/core/Grappa.sol

```diff
@@ -87,8 +91,36 @@ contract Grappa is Ownable, ReentrancyGuard {
     event MarginEngineRegistered(address engine, uint8 id);
     event OracleRegistered(address oracle, uint8 id);
 
-    constructor(address _optionToken) {
+    /*///////////////////////////////////////////////////////////////
+                Constructor for implementation Contract
+    //////////////////////////////////////////////////////////////*/
+
+    /// @dev set immutables in constructor
+    /// @dev also set the implemention contract to initialized = true
+    constructor(address _optionToken) initializer {
         optionToken = IOptionToken(_optionToken);
+        deployTimestamp = block.timestamp;
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                            Initializer
+    //////////////////////////////////////////////////////////////*/
+
+    function initialize() external initializer {
+        __Ownable_init();
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                    Override Upgrade Permission
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @dev Upgradable by the owner, can only be upgraded within 365 days after deployment
+     **/
+    function _authorizeUpgrade(
+        address /*newImplementation*/
+    ) internal view override onlyOwner {
```

Comment:
We should not use a modifier, they use more gas than calling a method.

Reply:
I think it's better to keep things this way here, just rely on the OZ library, and we won't be calling this often (hopefully)